xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -5266,14 +5267,18 @@ glusterd_import_friend_volumes_synctask(void *opaque)
     conf->restart_bricks = _gf_true;
 
     while (i <= count) {
-        /*Check if volume has any peer update */
-        update = (1UL & (arg->status_arr[(i / 64)] >> (i % 64)));
-        if (update) {
-            ret = glusterd_import_friend_volume(peer_data, i, arg);
-            if (ret)
+        bm = arg->status_arr[i / 64];
+        while (bm != 0) {
+            mask = bm &
+                   (-bm); /* mask will contain the lowest bit set from bm. */
```

Comment:
It would be better to put the comment above the line to avoid an ugly line split.